plugin-infra/go-plugin-access/src/main/java/com/thoughtworks/go/plugin/access/elastic/v5/ElasticAgentExtensionConverterV5.java

```diff
@@ -78,11 +78,14 @@ String getProfileViewResponseFromBody(String responseBody) {
         return new ImageDeserializer().fromJSON(responseBody);
     }
 
-    String getAgentStatusReportRequestBody(JobIdentifier identifier, String elasticAgentId) {
+    String getAgentStatusReportRequestBody(JobIdentifier identifier, String elasticAgentId, Map<String, String> clusterProfile) {
         JsonObject jsonObject = new JsonObject();
         if (identifier != null) {
             jsonObject.add("job_identifier", jobIdentifierJson(identifier));
         }
+        if (clusterProfile != null) {
```

Comment:
Not sure if this is the right place to do the null check.
`jsonObject.add("cluster_profile_properties", mapToJsonObject(clusterProfileProperties));`

can we instead send an empty map from the top, instead of sending null?